CMakePresets.json

```diff
@@ -0,0 +1,28 @@
+{
+    "version": 3,
+    "configurePresets": [
+        {
+            "name": "macOS base",
+            "hidden": true,
+            "binaryDir": "${sourceDir}/build/",
+            "generator": "Xcode",
```

Comment:
This should probably be labelled "xcode", otherwise it's too easy to pick it in e.g. vscode and then have a non-functioning build setup.